build.cake

```diff
@@ -128,17 +128,26 @@ Task("Build")
         DotNetBuild(SOLUTION_FILE, CreateDotNetBuildSettings());
     });
 
-DotNetBuildSettings CreateDotNetBuildSettings() =>
-    new DotNetBuildSettings
+DotNetBuildSettings CreateDotNetBuildSettings() 
+{
+    var version = packageVersion.ToString(); 
+    var assemblyVersion = version.Substring(0, version.LastIndexOf('-')) + ".0";
```

Comment:
The `version` only has a `-` if there is one in the tag. It won't be there for release builds causing this to fail.
The [MinVerVersion](https://www.cakebuild.net/api/Cake.MinVer/MinVerVersion/) class has a property `FileVersion` that does seem to do what you want here.

Reply:
I'm not sure why you set `AssemblyVersion` to the same as `FileVersion`?
That way code doesn't see a difference between any packages created between releases.

Reply:
The Minver FileVersion and Assemblyversion is by default not quite what I believe we want.  We could do it way, but we have never discussed it.  So I therefore just made them equal.  Minver is however not in line with the guidance they point to themselves at https://github.com/adamralph/minver?tab=readme-ov-file#version-numbers to https://learn.microsoft.com/en-ca/dotnet/standard/library-guidance/versioning#version-numbers.

One option is to keep the assembly version the way it is now, 3 first digits set and 4th = 0.  Then let FileVersion be with the 4th digit, which is normally the number of commits from the last tag, otherwise 0.

I am open to anything here actually, as long as we set them to something meaningful, and which matches the package version.

Reply:
This is the latest results:  
![image](https://github.com/nunit/nunit/assets/203432/c0bffc1d-21ec-45ce-99f5-35edac4d0875)